src/sentry/grouping/ingest/config.py

```diff
@@ -27,7 +38,9 @@ def _project_should_update_grouping(project: Project) -> bool:
     should_update_org = (
         project.organization_id % 1000 < float(settings.SENTRY_GROUPING_AUTO_UPDATE_ENABLED) * 1000
     )
-    return bool(project.get_option("sentry:grouping_auto_update")) and should_update_org
+    deprecated_config = project.get_option("sentry:grouping_config") in CONFIGS_TO_DEPRECATE
+    auto_update = bool(project.get_option("sentry:grouping_auto_update"))
+    return deprecated_config or (auto_update and should_update_org)
```

Comment:
Nit: This makes it clearer the variable isn't pointed to a config itself.

```suggestion
    is_deprecated_config = project.get_option("sentry:grouping_config") in CONFIGS_TO_DEPRECATE
    auto_update = bool(project.get_option("sentry:grouping_auto_update"))
    return is_deprecated_config or (auto_update and should_update_org)
```